RV, Glide

h1node opened the pull request.

h1node committed:
RV
Glide

kekulta commented on app/build.gradle.kts:
```diff
@@ -36,7 +36,8 @@ android {
 
 dependencies {
-
+    implementation(libs.glide)
+    implementation(libs.androidx.material)
```
Здорово, что используешь version catalog!

kekulta commented on app/src/main/java/com/playlistmaker/SearchActivity.kt:
```diff
@@ -12,22 +12,68 @@ import android.widget.ImageView
+
+        val onItemClickListener = object : OnItemClickListener {
+            override fun click(track: Track) {
+                TODO("Not yet implemented")
```
Коммитить код с функцией TODO() плохая идея. 
Можно где-то случайно забыть убрать и приложение упадёт. 
Конкретно здесь, лучше показывать тост либо вообще ничего не делать.

kekulta commented on app/src/main/java/com/playlistmaker/SettingsActivity.kt:
```diff
@@ -19,6 +20,24 @@ class SettingsActivity : AppCompatActivity() {
+        supportActionBar?.setDisplayHomeAsUpEnabled(true)
+        supportActionBar?.setHomeButtonEnabled(true)
+
+        sendEmail()
+        switchThemes()
+        share()
+        userAgreement()
```
Не очень хороший нейминг функций. По названиям кажется, то они отправляют емейл/переключают тему/etc, когда по факту они настраивают кнопки.

kekulta commented on app/src/main/java/com/playlistmaker/view/rv_viewholder/MusicViewHolder.kt:
```diff
@@ -0,0 +1,29 @@
+import com.playlistmaker.Track
+
+
+class MusicViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
+
+    private val trackName: TextView = itemView.findViewById(R.id.music_title)
+    private val artistName: TextView = itemView.findViewById(R.id.artist_name)
+    private val trackTime: TextView = itemView.findViewById(R.id.time)
+    private val poster: ImageView = itemView.findViewById(R.id.poster)
```
Лучше использовать ViewBinding.

kekulta commented on app/src/main/java/com/playlistmaker/view/rv_adapter/MusicRVAdapter.kt:
```diff
@@ -0,0 +1,36 @@
+    override fun getItemCount() = items.size
+}
+
+interface OnItemClickListener {
+    fun click(track: Track)
+}
```
Можно использовать fun interface либо лямбду.

h1node committed:
RV Glide ViewBinding
try to make correct layouts

h1node committed:
try to make correct layouts again
- инсеты расставил, картинку поменял
- закругление углов плейсхолдера увеличил
 .transform(RoundedCorners(dpToPx(10f, itemView.context))) если это не коректно, то можно отдельный стиль поставить
